File: presentationmanager/src/main/java/com/fyp/presentationmanager/model/EvaluationFormModel.java
package com.fyp.presentationmanager.model;

import com.fyp.presentationmanager.enums.EvaluationType;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class EvaluationFormModel {
    private Integer id;
    private Integer scheduleId;
    private EvaluationType evaluationType;
    private Integer maxGap;
    private Integer rubricUrl;
    private List<CriteriaModel> criteriaModels;
}

File: presentation-manager-web/src/app/model/evaluation-form.model.ts
import {CriteriaModel} from './criteria.model';

export class EvaluationFormModel {
  id: number;
  scheduleId: number;
  evaluationType: EvaluationType;
  maxGap: number;
  rubricUrl: string;
  criteriaModels: CriteriaModel[];
}

export enum EvaluationType {
  // PRESENTATION = 'Presentation',
  // REPORT = 'Report',
  // PANEL = 'Panel',
  // CHAIRPERSON = 'Chairperson'
  PRESENTATION = 'PRESENTATION',
  REPORT = 'REPORT',
  PANEL = 'PANEL',
  CHAIRPERSON = 'CHAIRPERSON'
}

export enum EvaluationFormMode {
  VIEW = 'View',
  EDIT = 'Edit',
  EVALUATE = 'Evaluate'
}

